src/main/java/org/cloudfoundry/autosleep/worker/remote/CloudFoundryApi.java

```diff
@@ -101,46 +112,39 @@ public void onSubscribe(Subscription subscription) {
     @Autowired
     private SpringLoggingClient logClient;
 
-    @Override
-    public void bindServiceInstance(ApplicationIdentity application, String serviceInstanceId) throws
-            CloudFoundryException {
-        log.debug("bindServiceInstance");
-        try {
-            cfClient.serviceBindings()
-                    .create(
-                            CreateServiceBindingRequest
-                                    .builder()
-                                    .applicationId(application.getGuid())
-                                    .serviceInstanceId(serviceInstanceId)
-                                    .build())
-                    .get(Config.CF_API_TIMEOUT);
-        } catch (RuntimeException r) {
-            throw new CloudFoundryException(r);
-        }
+    private <T, U> void bind(List<T> objectsToBind, Function<T, Mono<U>> caller)
+            throws CloudFoundryException {
+        log.debug("bind - {}", objectsToBind.size());
+        final CountDownLatch latch = new CountDownLatch(objectsToBind.size());
+        final AtomicReference<Throwable> errorEncountered = new AtomicReference<>(null);
+        final Subscriber<U> subscriber
+                = new BaseSubscriber<>(latch, errorEncountered::set, null);
+
+        objectsToBind.forEach(objectToBind -> caller.apply(objectToBind).subscribe(subscriber));
+        waitForResult(latch, errorEncountered, null);
     }
 
     @Override
-    public void bindServiceInstance(List<ApplicationIdentity> applications, String serviceInstanceId) throws
+    public void bindApplications(String serviceInstanceId, List<ApplicationIdentity> applications) throws
```

Comment:
I do not agree: if you look at the code, the two resources that are bound are either a service dans an application, or a service and a route. Truely, he route belongs to an application, but it is an other entity.